### PR TITLE
PR-2360: Create cse ios sdk 

### PR DESCRIPTION
**Pull request by ngub**

### CODE REVIEW

File: PayrailsCSE/LICENSE [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Copyright (c) 2023 Nikita Gubchenko <nikita@payrails.com>
```
Comment (vincentdaniel) [inline PayrailsCSE/LICENSE]: Let's check with platform team but I would rather put a [default MIT license](https://github.com/payrails/web-cse/blob/main/LICENSE)

File: PayrailsCSE/PayrailsCSE.podspec [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#
+# Be sure to run `pod lib lint PayrailsCSE.podspec' to ensure this is a
```
Comment (vincentdaniel) [inline PayrailsCSE/PayrailsCSE.podspec]: Can we add that to github ci?

File: PayrailsCSE/PayrailsCSE.podspec [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  s.swift_version = '5.8'
+
+  s.description      = <<-DESC
+SDK provides client side encryption and tokenization on Payrails platform.
+                       DESC
+
+  s.homepage         = 'https://github.com/payrails/ios-cse'
+  s.license          = { :type => 'MIT', :file => 'LICENSE' }
+  s.author           = { 'Nikita Gubchenko' => 'nikita@payrails.com' }
```
Comment (vincentdaniel) [inline PayrailsCSE/PayrailsCSE.podspec]: Can we make it non personal? see with #platform if we have a group that can be used instead

File: PayrailsCSE/Sources/PayrailsCSE/PayrailsCSE.swift [hunk 4]
```diff
@@ -0,0 +1,273 @@
+import Foundation
+import JOSESwift
+
+struct InitResponse: Codable {
+    let version: String
+    let data: String
+}
+
+struct Card: Codable {
+    let holderName: String?
```
Comment (vincentdaniel) [inline PayrailsCSE/Sources/PayrailsCSE/PayrailsCSE.swift]: Holder reference is also mandatory now here:
```suggestion
    let holderReference: String
    let holderName: String?
```

File: PayrailsCSE/Sources/PayrailsCSE/PayrailsCSE.swift [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+    let holderReference: String
+    let encryptedInstrumentDetails: String
+    let futureUsage: FutureUsage?
+    let storeInstrument: Bool?
+}
+
+enum FutureUsage: String, Codable {
+    case recurring
+    case cardOnFile
+    case unscheduledCardOnFile
```
Comment (vincentdaniel) [inline PayrailsCSE/Sources/PayrailsCSE/PayrailsCSE.swift]:
```suggestion
    case Subscription
    case CardOnFile
    case UnscheduledCardOnFile
```

File: PayrailsCSE/Sources/PayrailsCSE/PayrailsCSE.swift [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+    case transient
+}
+
+enum PaymentMethodType: String, Codable {
+    case card
+    case applePay
+    case googlePay
+    case klarna
+    case klarnaPaynow
+    case klarnaAccount
```
Comment (vincentdaniel) [inline PayrailsCSE/Sources/PayrailsCSE/PayrailsCSE.swift]:
```suggestion
    case klarna_paynow
    case klarna_account
```

File: PayrailsCSE/Tests/PayrailsCSETests/PayrailsCSETests.swift [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import XCTest
+@testable import PayrailsCSE
+
+final class PayrailsCSETests: XCTestCase {
+    func testExample() throws {
+        // This is an example of a functional test case.
+        // Use XCTAssert and related functions to verify your tests produce the correct
+        // results.
+        XCTAssertEqual(PayrailsCSE().text, "Hello, World!")
```
Comment (vincentdaniel) [inline PayrailsCSE/Tests/PayrailsCSETests/PayrailsCSETests.swift]: I wouldn't keep the test if it's not meaningful

**Review by vincentdaniel — DISMISSED** [review]

File: README.md [hunk 8]
```diff
@@ -1,19 +1,5 @@
...
-...
-
-### Reporting Vulnerabilities
-
-...
-
-### Contributing
-
-...
+SDK provides client side encryption and tokenization on Payrails platform.
```
Comment (in2lag) [inline README.md:5]: should we write docs with top level API? as it's supposed to be public repo

File: PayrailsCSE/LICENSE (hunk 1 above)
Comment (ngub) [reply PayrailsCSE/LICENSE]: Oops 😂

**Review by in2lag — APPROVED** [review]

**Pull request merged**